### ISSUE TITLE
Go Swagger Annotation Considered deadcode

### ISSUE THREAD

**Issue by shanedetsch** [golangci/golangci-lint#1304]
**The golangci-lint deadcode lint incorrectly considers swagger annotation as dead code.**

The golangci-lint version

```
# golangci-lint --version
golangci-lint has version 1.30.0 built from 45b90f6 on 2020-08-03T03:10:34Z
```

The golangcli-lint validation failure 

```
  {
    "description": "deadcode: `appSwaggerParameterDefinition` is unused",
    "fingerprint": "A4AB72*********E99AEAA3D",
    "location": {
      "path": "app/server.go",
      "lines": {
        "begin": 110
      }
    }
  },
```

However this is not dead code as it is a type of swagger annotation. 

```
// DEFINED HERE
// Swagger parameters for the post endpoint
// swagger:parameters SwaggerParameterDefinition
type appSwaggerParameterDefinition struct {
	// The POST body for creating an Item
	// in: body
	// required: true
	Body service.RequestItemForPost
	// The integration type
	// in: path
	// required: true
	// example: {genericType}
	itemType string `json: "itemType"`
}

// USED HERE
// Swagger Docs for the post sql server endpoint where SwaggerParameterDefinition is a link to the Swagger parameters for this endpoint
// swagger:route POST /{itemType}/endpoint SwaggerParameterDefinition
// Returns created Item
// Schemes: http
// Responses:
//  200: okResponse
//  500: errorResponse
// handlePostItem creates an Item
func (s *IntegrationService) handlePostItem(c echo.Context) error {

```

**Comment by shanedetsch** [comment]
Work around

```
+// nolint:deadcode
 type appSwaggerParameterDefinition struct {
        // The POST body for creating an Item
```

**Comment by SVilgelm** [comment]
It's not mentioned anywhere, you need to use it somewhere in the code. Now the `deadcode` linter show the correct message, the `appSwaggerParameterDefinition` is unused in provided sample

I would suggest you to use it in the response:
```go
func (s *IntegrationService) handlePostItem(c echo.Context) error {
var doc appSwaggerParameterDefinition
c.Bind(&doc)
...
}
```

**Comment by shanedetsch** [comment]
@SVilgelm  Your solution is clever, however I think Go Swagger ties the annotation in the handler and struct definition so instantiating the struct from the context in the handler isn't necessary.

**Comment by SVilgelm** [comment]
nope, it's not doing any magic,  `appSwaggerParameterDefinition` should be used in generated code then. 

If you have a generated code and that code uses the `appSwaggerParameterDefinition` and the dead code ignores that usage, only then it's a bug

otherwise you have a definition, but go does not use it and it probably uses map[string]interface{} instead of the `appSwaggerParameterDefinition` struct

**Comment by shanedetsch** [comment]
Well the code is not used in production however the Swagger annotation is used by the ( swagger ) cli which will use a code ( model scan ) to generate apidocs.  Thus maybe the best solution is to just use ( nolint:deadcode ).

**Comment by SVilgelm** [comment]
actually you just store the definitions and uses them I another project. In this case yes you are right, using the `nolint:deadcode` is a solution, but it's definitely not a bug of the `deadcode` linter

**Comment by shanedetsch** [comment]
Yeah right probably not a ( deadcode ) bug I see now.

**Comment by SVilgelm** [comment]
so. I'm closing the issue

**Comment by shanedetsch** [comment]
Yeah close, hopefully other people who are confused by swaggers interaction with deadcode can find this thread useful and use ( nolint:deadcode )
